### PR TITLE
[OBSDEF-3628] Signpost component enhancement

### PR DESCRIPTION
**Pull request by ghatea**
Description: 
This PR includes enhancement for signpost component. Custom view support has been added where use can pass any custom view from which user wants to open popup view. Also updated x, y transformations calculations of popover view.

Screenshot:
Screenshot (441)

### CODE REVIEW

File: src/forms/signpost/SignPost.tsx [hunk 1]
```diff
@@ -39,6 +40,7 @@ type SignPostProps = {
     onClick?: Function;
     onClose?: Function;
     dataqa?: string;
+    showCustomOpenAt?: boolean;
```
Comment (akate1) [inline src/forms/signpost/SignPost.tsx:43]: `showCustomOpenAt -> customSignpost`? `openAt `is different prop used for different purpose please don't mix this here

File: src/forms/signpost/SignPost.tsx [hunk 2]
```diff
@@ -156,8 +158,13 @@ export class SignPost extends React.PureComponent<SignPostProps> {
 
     buildSignPostTriggerButton = () => {
         const {isOpen} = this.state;
-        const {openAt} = this.props;
-        return (
+        const {openAt, showCustomOpenAt} = this.props;
+        const className = isOpen ? "signpost-action signpost-trigger active" : "signpost-action signpost-trigger";
```
Comment (akate1) [inline src/forms/signpost/SignPost.tsx:162]: Please create classNames file or constants for these classes

**Review by akate1 — COMMENTED** [review]
@ghatea Please update version on package.json

File: src/forms/signpost/SignPost.stories.tsx [hunk 3]
```diff
@@ -161,4 +161,18 @@ storiesOf("Signposts", module)
                 </SignPost>
             </div>
         </div>
+    ))
+    .add("SingPost with custom view", () => (
```
[the remark this answers is not in the document]
Comment (ghatea) [reply src/forms/signpost/SignPost.stories.tsx]: changed the story name to custom SignPost

File: src/forms/signpost/SignPost.tsx (hunk 1 above)
Comment (ghatea) [reply src/forms/signpost/SignPost.tsx]: changed the prop name to 'customSignPost'

File: src/forms/signpost/SignPost.tsx (hunk 2 above)
Comment (ghatea) [reply src/forms/signpost/SignPost.tsx]: created className constants.

File: src/forms/signpost/SignPost.tsx [hunk 4]
```diff
@@ -14,6 +14,10 @@ import {Button} from "../button";
 import {Icon, IconProps} from "../../icon";
 import {calculateAxisPosition} from "./AxisPosition";
 
+//className constants
+const signPostActive = "signpost-action signpost-trigger active";
```
Comment (akate1) [inline src/forms/signpost/SignPost.tsx:18]: Please create different constants for these 3 classes and use ClassNames() in utils to assign array of class names to div/button

Comment (ghatea) [reply src/forms/signpost/SignPost.tsx]: Done.

File: src/forms/signpost/SignPost.tsx [hunk 5]
```diff
@@ -156,16 +163,20 @@ export class SignPost extends React.PureComponent<SignPostProps> {
...
     buildSignPostTriggerButton = () => {
         const {isOpen} = this.state;
-        const {openAt} = this.props;
-        return (
+        const {openAt, customSignPost} = this.props;
+
+        return customSignPost ? (
+            <div
+                onClick={this.handleOnClick}
+                className={classNames([signPostAction, signPostTrigger, isOpen && signPostActive])}
```
Comment (akate1) [inline src/forms/signpost/SignPost.tsx:171]: Please add className prop as well for user defined CSS classnames.

File: src/forms/signpost/SignPost.tsx [hunk 6]
```diff
@@ -156,16 +163,20 @@ export class SignPost extends React.PureComponent<SignPostProps> {
...
+        ) : (
             <Button
-                className={classNames([
-                    "signpost-action", //prettier
-                    "btn-small",
-                    "btn-link",
-                    "signpost-trigger",
-                    isOpen && "active",
-                ])}
+                className={classNames([signPostAction, signPostTrigger, isOpen && signPostActive])}
```
Comment (akate1) [inline src/forms/signpost/SignPost.tsx:177]: you can create single constant for classNames and use it here and above line like,
const signPostClasses = classNames([signPostAction, signPostTrigger, isOpen && signPostActive, <Prop value for user defined className>])

File: src/forms/signpost/SignPost.tsx [hunk 7]
```diff
@@ -29,6 +34,8 @@ import {calculateAxisPosition} from "./AxisPosition";
  *@param {onClick} callback function to call on open of signPost
  *@param {onClose} callback function to call on close of signPost
  *@param {dataqa} for Quality Engineering
+ *@param {customSignPostTrigger} flag to show custom signpost trigger
+ *@param {signPostClassNames} css classNames for signPost
```
Comment (akate1) [inline src/forms/signpost/SignPost.tsx:38]: signPostClassNames -> triggerClassNames

**Review by akate1 — APPROVED** [review]
LGTM. Left 1 comment

**Review by maheshk172 — APPROVED** [review]
lgtm

**Pull request merged**